src/arrays.jl

```diff
@@ -156,10 +156,10 @@ end
 materialize_first(bc::Base.Broadcast.Broadcasted) = bc.f(materialize_first.(bc.args)...)
 
 # Base cases
-materialize_first(q::AbstractQuantity{<:AbstractArray}) = new_quantity(typeof(q), first(ustrip(q)), dimension(q))
-materialize_first(q::AbstractQuantity) = q
+materialize_first(q::AbstractGenericQuantity{<:AbstractArray}) = new_quantity(typeof(q), first(ustrip(q)), dimension(q))
```

Comment:
I wonder what our level of support of array-containing `Quantity`'s is, e.g. do we expect users to ever use them in the wild, or just something we support because we can, and we can ponder their use as time goes on? They don't seem to offer anything more than `QuantityArray`s (although maybe they could replace them -- they do at least already carry the information that allows the compiler to in theory lift out dimension calculations)